### PR TITLE
Add a Dart package example to `devtools_extensions/example`

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR adds an example of a pure Dart package providing a DevTools extension to `devtools_extensions/example`. 

- Create a new example `devtools_extensions/example/packages_with_extensions/dart_foo` with the following structure:
```
dart_foo/
  packages/
    dart_foo/   # New pure Dart package
      extension/
        build/  # DO NOT REVIEW - generated build output
        config.yaml
        .pubignore
    dart_foo_devtools_extension/   # New Flutter Web app
```
- use `package:dart_foo` from the existing example app, `app_that_uses_foo` under `bin`, and also under `test`.
- Add VS code configurations to run these extensions and the respective example app usages.

This PR is not as big as it looks - you don't need to review the code under `dart_foo/extension/build` (this is generated build output of `dart_foo_devtools_extension`), and `dart_foo_devtools_extension` is a new Flutter web app, so there are files surrounding the main.dart file there that are necessary for creating a new flutter web app package.

### CODE REVIEW

File: packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo/lib/dart_foo.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo/lib/dart_foo.dart:1]: review

File: packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: dart_foo
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo/pubspec.yaml:1]: review

File: packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# dart_foo_devtools_extension
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/README.md:1]: review

File: packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/lib/main.dart:1]: review

File: packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+name: dart_foo_devtools_extension
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/example/packages_with_extensions/dart_foo/packages/dart_foo_devtools_extension/pubspec.yaml:1]: review

File: packages/devtools_extensions/example/app_that_uses_foo/test/dart_test/example_dart_test.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+// This test can be run to verify that the `package:foo` DevTools extension
+// loads properly when debugging a Dart test target with DevTools.
+//
+// This doubles as a test to make sure the DevTools extension loads properly
+// when the test target is in a subdirectory of 'test/'.
+//
+// To test this, run the following command and copy the VM service URI to
+// connect to DevTools:
+//
+// dart run test/dart_test/example_dart_test.dart --start-paused
```
Comment (bkonyi) [inline packages/devtools_extensions/example/app_that_uses_foo/test/dart_test/example_dart_test.dart:17]: What's using this `--start-paused` flag?

**Review by bkonyi — APPROVED** [review]

File: packages/devtools_extensions/example/app_that_uses_foo/test/dart_test/example_dart_test.dart (hunk 6 above)
Comment (kenzieschmoll) [reply packages/devtools_extensions/example/app_that_uses_foo/test/dart_test/example_dart_test.dart:17]: so we can open DevTools connected to the test and see the extension

**Pull request merged**